### PR TITLE
Replace Sapling balance and scan progress FFIs with wallet summary FFI

### PR DESCRIPTION
**Pull request by str4d**
Closes zcash/zcash-android-wallet-sdk#1282.

# Author

- [ ] **Self-review** your own code in GitHub's web interface[^1]
- [ ] Add **automated tests** as appropriate
- [ ] Update the [**manual tests**](../blob/main/docs/testing/manual_testing)[^2] as appropriate
- [ ] Check the **code coverage**[^3] report for the automated tests
- [ ] Update **documentation** as appropriate (e.g [README.md](../blob/main/README.md), [Architecture.md](../blob/main/docs/Architecture.md), etc.)
- [ ] **Run the demo app** and try the changes
- [ ] Pull in the latest changes from the **main** branch and **squash** your commits before assigning a reviewer[^4]

# Reviewer

- [x] Check the code with the [Code Review Guidelines](../blob/main/docs/CODE_REVIEW_GUIDELINES.md) **checklist**
- [x] Perform an **ad hoc review**[^5]
- [x] Review the **automated tests**
- [ ] Review the **manual tests**
- [ ] Review the **documentation**, [README.md](../blob/main/README.md), [Architecture.md](/blob/main/docs/Architecture.md), etc. as appropriate
- [x] **Run the demo app** and try the changes[^6]

[^1]: _Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs._
[^2]: _While we aim for automated testing of the SDK, some aspects require manual testing. If you had to manually test 
something during development of this pull request, write those steps down._
[^3]: _While we are not looking for perfect coverage, the tool can point out potential cases that have been missed. Code coverage can be generated with: `./gradlew check` for Kotlin modules and `./gradlew connectedCheck -PIS_ANDROID_INSTRUMENTATION_TEST_COVERAGE_ENABLED=true` for Android modules._
[^4]: _Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit._
[^5]: _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
[^6]: _While the CI server runs the demo app to look for build failures or crashes, humans running the demo app are 
more likely to notice unexpected log messages, UI inconsistencies, or bad output data. Perform this step last, after verifying the code changes are safe to run locally._

### CODE REVIEW

**Comment by str4d** [comment]
[Force-pushed](https://github.com/zcash/zcash-android-wallet-sdk/compare/ff8eed607898aafb8eac6ccc9bb9b2ab8a8cea71..283794eb91c4c3d7be986f91a08af5a1228ee418) to fix lints.

**Comment by str4d** [comment]
[Force-pushed](https://github.com/zcash/zcash-android-wallet-sdk/compare/283794eb91c4c3d7be986f91a08af5a1228ee418..de8ee0e4a808c3678a06445c04fb5af838f9ab9e) to fix tests (which apparently weren't being compiled by the default Android Studio build pathway).

**Comment by str4d** [comment]
[Force-pushed](https://github.com/zcash/zcash-android-wallet-sdk/compare/de8ee0e4a808c3678a06445c04fb5af838f9ab9e..6f0e6ccfaf795871c8c524af11913bc7f597e208) to fix more tests.

File: backend-lib/src/main/rust/lib.rs [hunk 1]
```diff
@@ -1147,25 +1084,64 @@ pub unsafe extern "C" fn Java_cash_z_ecc_android_sdk_internal_jni_RustBackend_ge
...
+    balance: &AccountBalance,
+) -> jni::errors::Result<JObject<'a>> {
+    let sapling_total_balance = Amount::from(balance.sapling_balance.total());
+    let sapling_verified_balance = Amount::from(balance.sapling_balance.spendable_value);
+
+    env.new_object(
+        JNI_ACCOUNT_BALANCE,
+        "(IJJ)V",
+        &[
+            JValue::Int(u32::from(*account) as i32),
```
Comment (daira) [inline backend-lib/src/main/rust/lib.rs:1101]:
```suggestion
            JValue::Int(u32::from(*account).try_into().unwrap()),
```

**Review by daira — CHANGES_REQUESTED** [review]
utACK modulo the issue with encoding an `AccountId` as a Java/Kotlin `int`.

File: backend-lib/src/main/rust/lib.rs (hunk 1 above)
Comment (str4d) [reply backend-lib/src/main/rust/lib.rs:1101]: That is a problem with the `AccountId` type, that should be separately fixed. The rest of the account logic enforces the correct `u31` range, in particular [`WalletDb::create_account`](https://github.com/zcash/librustzcash/blob/e6fa5673328523553e6aa7e083c3fbeaa8dca6fe/zcash_client_sqlite/src/lib.rs#L378-L384). And [casting `u32` to `i32` is a no-op](https://doc.rust-lang.org/reference/expressions/operator-expr.html#numeric-cast), so for the `u31` range that account IDs are in, this is exactly what we want.

Comment (daira) [reply backend-lib/src/main/rust/lib.rs:1101]: As I said, if account ids are required to be 31 bits, then this is preferable:
```suggestion
            JValue::Int(u32::from(*account).try_into().unwrap()),
```

**Review by daira — APPROVED** [review]
utACK modulo comment.

**Comment by str4d** [comment]
[Force-pushed](https://github.com/zcash/zcash-android-wallet-sdk/compare/6f0e6ccfaf795871c8c524af11913bc7f597e208..8689375b68c508c1db311f6546568b1355e9076c) to fix a JNI type signature.

File: backend-lib/src/main/rust/lib.rs (hunk 1 above)
Comment (str4d) [reply backend-lib/src/main/rust/lib.rs:1101]: From a semantic perspective these are equivalent. From a codegen perspective, the suggestion results in a bunch of extra code branches that will never be taken, but can't be compiled out. And from a practical perspective, this is already done elsewhere in the FFI, so this is style consistency.

**Review by HonzaR — APPROVED** [review]
Thank you for these changes! It looks good to me. I just pushed a few minor documentation and testing improvements.

**Pull request merged**